### DISCUSSION TITLE
Common code for different modules

### DISCUSSION THREAD

**Discussion by LuisVilaplana** [oqtane/oqtane.framework#4289] [General]
What is the best practice for creating common code for different modules in Oqtane? I am developing several modules that use the same models, how can I avoid repeating the declaration of those classes in each module? Can a Class Library be created and referenced in the Shared project of each module?

**Comment by leigh-pointer** [comment]
Currently my personal preference is to add classes and components as linked references. I have all my reusable code files off in another repository.

**Reply by LuisVilaplana** [reply]
Ok, that sounds good, I'm going to do it that way too. Thank you

**Comment by lanthonyneville** [comment]
You could also look at creating a "library" module (see #4192). The feature Shaun added to have these recognised in the framework will be in a future release, but the approach described in the accepted answer works well (this is what I am doing).
